staging/src/k8s.io/apimachinery/pkg/api/validate/schema.go

```diff
@@ -72,3 +74,16 @@ func Optional[T comparable](_ operation.Context, fldPath *field.Path, value, _ *
 	}
 	return field.ErrorList{field.Required(fldPath, "optional value was not specified")}
 }
+
+// MaxItems verifies that the specified slice is not longer than max items.
+func MaxItems[T any](_ operation.Context, fldPath *field.Path, value, _ T, max int) field.ErrorList {
+	size := 0
+
+	if reflect.TypeOf(value).Kind() == reflect.Slice {
```

Comment:
I don't understand this - why not just take a `[]T` argument.  If someone tags a field that is not a slice, it will fail to compile, which is what we want.  Right?